### PR TITLE
DBZ-1419 Support for Connect temporal types in SQL Server

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java [hunk 1]
```diff
@@ -219,12 +220,25 @@ public static DecimalHandlingMode parse(String value, String defaultValue) {
                                                       .withInvisibleRecommender()
                                                       .withDescription("The schemas for which events must not be captured");
 
+    public static final Field TIME_PRECISION_MODE = Field.create("time.precision.mode")
+                                                             .withDisplayName("Time Precision")
+                                                             .withEnum(TemporalPrecisionMode.class, TemporalPrecisionMode.ADAPTIVE)
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java]: @jpechane, doesn't this make `adaptive` available to MySQL again and, worse, make it the default for it, too?

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java]: There is `MySqlConnectorConfig#TIME_PRECISION_MODE` which doesn't contain the value; going forward, we should remove that and use the one from the base, but validate that `adaptive` isn't used (code for that is in place already).

Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseConnectorConfig.java]: @gunnarmorling Fixed, I forgot to override the attribute/method.

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerValueConverters.java [hunk 2]
```diff
@@ -43,9 +43,11 @@ public SqlServerValueConverters() {
      *            treated; may be null if
      *            {@link io.debezium.jdbc.JdbcValueConverters.DecimalMode#PRECISE}
      *            is to be used
+     * @param temporalPrecisionMode
+     *            date/time value will be represented either as Connect datatypes or Debezium specific datatypes
      */
-    public SqlServerValueConverters(DecimalMode decimalMode) {
-        super(decimalMode, TemporalPrecisionMode.ADAPTIVE_TIME_MICROSECONDS, ZoneOffset.UTC, null, null);
+    public SqlServerValueConverters(DecimalMode decimalMode, TemporalPrecisionMode temporalPrecisionMode) {
```
Comment (gunnarmorling) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerValueConverters.java]: So this causes the default to change from ADAPTIVE_TIME_MICROSECONDS to ADAPTIVE; I think that makes sense though (should have been like that before), so I'll only add it to the change notes.

**Comment by Naros** [comment]
Rebased against master to pull in DBZ-777 fixes to overlay a couple test fixture changes observed to still fail during CI jobs.  @gunnarmorling This should be ready for you if CI passes.

**Comment by gunnarmorling** [comment]
Sweet, thanks; @jpechane and @Naros!

**Pull request merged**